proxy/proxy.go

```diff
@@ -1526,3 +1529,35 @@ func ensureUTF8(s string) string {
 	}
 	return fmt.Sprintf("invalid utf-8: %q", s)
 }
+
+// Opt-In for filters to get called Response(ctx) in case of errors
+type errorHandlerFilter interface {
+	HandleErrorResponse()
+}
+
+func (p *Proxy) handleFiltersOnError(ctx *context, filters []*routing.RouteFilter) {
+	filtersStart := time.Now()
+	filterTracing := p.tracing.startFilterTracing("response_filters", ctx)
+	defer filterTracing.finish()
+
+	last := len(filters) - 1
+	for i := range filters {
+		fi := filters[last-i]
```

Comment:
Not sure why it was done like that before, I'd do:
```go
for i:=len(filters)-1; i>=0; i-- {
  fi := filters[i]
}
```

Reply:
I can change both to match it, but I think we should have one style.

Reply:
done